sympy/core/add.py

```diff
@@ -576,6 +580,8 @@ def _eval_is_positive(self):
             v = _monotonic_sign(a)
             if v is not None:
                 s = v + c
+                if s == self:
+                    return None
                 if s.is_positive and a.is_nonnegative:
```

Comment:
The check here is wrong (we are testing both `s.is_positive` and `a.is_negative`, so there isn't necessarily recursion). 

But I'm having a hard time understanding the `_monotonic_sign` logic. @smichr can you explain what this logic is supposed to do, and what the correct behavior should be if `s` is the same as `self`?